### ISSUE TITLE
terminal newline detection is LANGUAGE-dependent

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#662]
Noticed because of #659 :+1: 

```
LANGUAGE=zh_CN Rscript -e "testthat::test_dir('tests')"
── 1. Failure: Terminal newlines are detected correctly (@test-get_source_expressions.R#75)  ─────────────────────────────────────────────
get_source_expressions(tmp2)$expressions[[2L]]$terminal_newline isn't false.
```

That's because the linter depends on matching a message in English:

```
grepl("incomplete final line found on", w$message, fixed = TRUE)
```

We can probably use `gettext` instead:

```
gettextf("incomplete final line found on '%s'", 1, domain="R")
# [1] "读'%s'时最后一行未遂"
```

**Comment by AshesITR** [comment]
Dupe of #661
